doc/source/whatsnew/v1.0.0.rst

```diff
@@ -1108,6 +1108,7 @@ Numeric
 - Bug in :meth:`DataFrame.round` where a :class:`DataFrame` with a :class:`CategoricalIndex` of :class:`IntervalIndex` columns would incorrectly raise a ``TypeError`` (:issue:`30063`)
 - Bug in :meth:`Series.pct_change` and :meth:`DataFrame.pct_change` when there are duplicated indices (:issue:`30463`)
 - Bug in :class:`DataFrame` cumulative operations (e.g. cumsum, cummax) incorrect casting to object-dtype (:issue:`19296`)
+- Bug in dtypes being lost in ``DataFrame.__invert__`` (``~`` operator) with mixed dtypes (:issue:`31183`)
```

Comment:
Should these notes not be in 1.0.1 to match the other PR?

Reply:
My plan was to just let the other PR also be backported (since it is tagged with the milestone), then I don't need to move it here as well

Reply:
Makes sense